cpp/unit/unit_texture.cpp

```diff
@@ -145,7 +145,7 @@ unsigned int dir_group(coord::phys3_delta dir, unsigned int angles) {
 	unsigned int first_angle = 5 * angles / 8;
 
 	// normalise direction vector
-	double len = std::hypot(dir.ne, dir.se);
+	double len = std::hypot(dir.ne, dir.se).to_double();
```

Comment:
I think having different precisions as result type would complicate this a lot. Isn't it good enough to keep everything a `phys_t`?